the-collab-lab/tcl-77-smart-shopping-list
# Fix Prod Build Bug w/ Bootstrap & Updated Dependencies

## Description



As we discovered on Monday the production version of the app was mad 😡. It was working locally but not after getting built somehow. While the build steps were "passing" in Github Actions something was not right. 

In office hours we updated some of the dependencies, but the build still wasn't working. Then we looked at how `bootstrap/react-bootstrap` worked with `vite` and how `sass` needed to be imported. We wrote a resolver in the `vite config` file to be able to properly resolve the `sass` imports as we found in the documentation. After we got to those steps we did some manual testing of how the application was handling the building of it in the `manualChunks` seeing that without that there wasn't an issue with the app but it was saying the minification was a bit large. 

After finding that the `manual chunking` seemed to be a sticking point, I went through the `vite` and `rollup` documentation to see how it was handled. With the error focusing on a `forwardRef` issue I did a bit of googling and it seemed it was something used by `react-bootstrap` and the `manual chunking` was chunking parts of the `node_modules` to minify it but was separating parts that the `react-bootstrap` needed in a dependency of it called `@restart` I added a `guard clause` to the `manual chunking` that would bundle the `react-bootstrap` and `@restart` into the same group. That for rid of the `forwardRef` and I got a `react minification #299` that when looking it up said the `target wasn't a DOM element` it was and changing the script tag in the html to `defer` got rid of that.

Lastly I noticed there was a duplication in the `AuthnticatedNavBar` and the one the app was using wasn't getting the `.scss` applied to that so I did correct that here as well.

## Acceptance Criteria


- [x] MAKE THE APP LOAD AGAIN 😅😂🎉

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | ⬆️   update dependencies |
|  ✓    | 🐛   bug fix |
|  ✓    | 💚   fix prod build |


## Updates

### Before



<img width="1624" alt="Screenshot 2024-10-04 at 3 34 34 AM" src="https://github.com/user-attachments/assets/d00492a5-b9fc-4f20-8f34-d95420a0e494">

### After


<img width="1624" alt="Screenshot 2024-10-04 at 3 39 30 AM" src="https://github.com/user-attachments/assets/2eafb586-6c81-49c5-887d-b901155ccd2b">


## Testing Steps / QA Criteria

- Please click the [preview link](https://tcl-77-smart-shopping-list--pr59-update-deps-d8hg77f6.web.app) of this PR the issue has been only happen in the prod app after it builds so need to confirm it works there.
